### PR TITLE
bluetooth: CS various updates

### PR DESCRIPTION
**Pull request by olivier-le-sage**
Various channel sounding-related changes/fixes

### CODE REVIEW

File: include/zephyr/bluetooth/hci_types.h [hunk 1]
```diff
@@ -3689,8 +3689,8 @@ struct bt_hci_le_cs_step_data_mode_1 {
 	uint8_t packet_nadm;
 	uint8_t packet_rssi;
 	union {
-		int16_t toa_tod_initiator;
-		int16_t tod_toa_reflector;
+		uint16_t toa_tod_initiator;
```
Comment (sean-madigan) [inline include/zephyr/bluetooth/hci_types.h:3692]: Is it worth some extra documentation with these?

**Review by theob-pro — DISMISSED** [review]

**Comment by alwa-nordic** [comment]
>  bluetooth: host: change CS ToX_ToX timestamps back to uint16_t
> 
> The special value 0x8000 is just outside the range of an int16_t
so to avoid potential signed overflow behavior issues, casting this
value is made the user's responsibility

Can you explain the signed overflow behavior issues?

**Comment by jhedberg** [comment]
> BT_HCI_LE_CS_TIME_DIFFERENCE_NOT_AVAILABLE is bigger than INT16_MAX. That's all really

Would it make sense to define the macro as `((int16_t)0x8000)'?

**Comment by olivier-le-sage** [comment]
> > BT_HCI_LE_CS_TIME_DIFFERENCE_NOT_AVAILABLE is bigger than INT16_MAX. That's all really
> 
> Would it make sense to define the macro as `((int16_t)0x8000)`?

instead of this change from int16_t to uint16_t, right ? I guess that's fine

**Comment by jhedberg** [comment]
> > Would it make sense to define the macro as `((int16_t)0x8000)`?
> 
> instead of this change from int16_t to uint16_t, right ? I guess that's fine

Yes, that's what I meant. That'd be more in line with how the HCI spec defines this parameter (it states that it's a signed integer).

**Review by jhedberg — APPROVED** [review]

**Review by theob-pro — APPROVED** [review]

**Pull request merged**